File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/workout/response/workout_response_dto.js
import {
    CustomerCollectionResponseDTO
} from "../../customer_collection/response/customer_collection_response_dto.js";

import {
    WorkoutTypeResponseDTO
} from "../../workout_type/response/workout_type_response_dto.js";

import {
    CustomerResponseDTO
} from "../../customer/response/customer_response_dto.js";

import {
    LangResponseDTO
} from "../../lang/response/lang_response_dto.js";

export class WorkoutResponseDTO {
    #id;
    #numberOfWords;
    #currentMilliseconds;
    #dateOfStart;
    #dateOfEnd;
    #customer;
    #workoutType;
    #langIn;
    #langOut;
    #customerCollection;

    constructor(json) {
        if (json) {
            this.#id = json["id"];
            this.#numberOfWords = json["number_of_words"];
            this.#currentMilliseconds = json["current_milliseconds"];

            let dateOfStart = json["date_of_start"];
            if (dateOfStart) {
                this.#dateOfStart = new Date(dateOfStart);
            }

            let dateOfEnd = json["date_of_end"];
            if (dateOfEnd) {
                this.#dateOfEnd = new Date(dateOfEnd);
            }

            let customer = json["customer"];
            if (customer) {
                this.#customer = new CustomerResponseDTO(customer);
            }

            let workoutType = json["workout_type"];
            if (workoutType) {
                this.#workoutType = new WorkoutTypeResponseDTO(workoutType);
            }

            let langIn = json["lang_in"];
            if (langIn) {
                this.#langIn = new LangResponseDTO(langIn);
            }

            let langOut = json["lang_out"];
            if (langOut) {
                this.#langOut = new LangResponseDTO(langOut);
            }

            let customerCollection = json["customer_collection"];
            if (customerCollection) {
                this.#customerCollection = new CustomerCollectionResponseDTO(customerCollection);
            }
        }
    }

    getId() {
        return this.#id;
    }

    getNumberOfWords() {
        return this.#numberOfWords;
    }

    getDateOfStart() {
        return this.#dateOfStart;
    }

    getDateOfEnd() {
        return this.#dateOfEnd;
    }

    getCurrentMilliseconds() {
        return this.#currentMilliseconds;
    }

    getCustomer() {
        return this.#customer;
    }

    getWorkoutType() {
        return this.#workoutType;
    }

    getLangIn() {
        return this.#langIn;
    }

    getLangOut() {
        return this.#langOut;
    }

    getCustomerCollection() {
        return this.#customerCollection;
    }
}

File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/workout/response/WorkoutResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.workout.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.json.l.LongDeserializer;
import ru.herooo.mylanguageweb.json.l.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.customer.response.CustomerResponseDTO;
import ru.herooo.mylanguageweb.dto.entity.customercollection.response.CustomerCollectionResponseDTO;
import ru.herooo.mylanguageweb.dto.entity.lang.response.LangResponseDTO;
import ru.herooo.mylanguageweb.dto.entity.workouttype.response.WorkoutTypeResponseDTO;

import java.time.LocalDateTime;

public class WorkoutResponseDTO {
    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    private long id;

    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_words")
    private long numberOfWords;

    @JsonProperty("date_of_start")
    private LocalDateTime dateOfStart;

    @JsonProperty("date_of_end")
    private LocalDateTime dateOfEnd;

    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("current_milliseconds")
    private long currentMilliseconds;

    @JsonProperty("customer")
    private CustomerResponseDTO customer;

    @JsonProperty("workout_type")
    private WorkoutTypeResponseDTO workoutType;

    @JsonProperty("lang_in")
    private LangResponseDTO langIn;

    @JsonProperty("lang_out")
    private LangResponseDTO langOut;

    @JsonProperty("customer_collection")
    private CustomerCollectionResponseDTO customerCollection;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getNumberOfWords() {
        return numberOfWords;
    }

    public void setNumberOfWords(long numberOfWords) {
        this.numberOfWords = numberOfWords;
    }

    public LocalDateTime getDateOfStart() {
        return dateOfStart;
    }

    public void setDateOfStart(LocalDateTime dateOfStart) {
        this.dateOfStart = dateOfStart;
    }

    public LocalDateTime getDateOfEnd() {
        return dateOfEnd;
    }

    public void setDateOfEnd(LocalDateTime dateOfEnd) {
        this.dateOfEnd = dateOfEnd;
    }

    public CustomerResponseDTO getCustomer() {
        return customer;
    }

    public void setCustomer(CustomerResponseDTO customer) {
        this.customer = customer;
    }

    public WorkoutTypeResponseDTO getWorkoutType() {
        return workoutType;
    }

    public void setWorkoutType(WorkoutTypeResponseDTO workoutType) {
        this.workoutType = workoutType;
    }

    public LangResponseDTO getLangIn() {
        return langIn;
    }

    public void setLangIn(LangResponseDTO langIn) {
        this.langIn = langIn;
    }

    public LangResponseDTO getLangOut() {
        return langOut;
    }

    public void setLangOut(LangResponseDTO langOut) {
        this.langOut = langOut;
    }

    public CustomerCollectionResponseDTO getCustomerCollection() {
        return customerCollection;
    }

    public void setCustomerCollection(CustomerCollectionResponseDTO customerCollection) {
        this.customerCollection = customerCollection;
    }

    public long getCurrentMilliseconds() {
        return currentMilliseconds;
    }

    public void setCurrentMilliseconds(long currentMilliseconds) {
        this.currentMilliseconds = currentMilliseconds;
    }
}
